src/net.h

```diff
@@ -506,7 +506,12 @@ class CNode
         // Don't relay addr messages to peers that we connect to as block-relay-only
         // peers (to prevent adversaries from inferring these links from addr
         // traffic).
-        return m_conn_type != ConnectionType::BLOCK_RELAY;
+        if (m_conn_type == ConnectionType::BLOCK_RELAY) return false;
+
+        // Don't relay addr messages to peers that have sent disabletx, as they are
+        // likely to be inbound block-relay-only peers.
+        if (m_disable_tx) return false;
```

Comment:
Moving `RelayAddrsWithConn` to net_processing and making it:

```c++
static inline bool RelayAddrsWithConn(const Node& node, const Peer& peer) {
    return !node.IsBlockOnlyConn() && !peer.m_disable_tx;
}
```

would preserve the net/net_processing separation better, I think?

Reply:
I think that without a bigger refactoring, John's observation that we call into `ForEachNode` but access data in `net_processing` would still remain.  Given that we have a major refactoring effort underway (and I think #21186 will address this directly, from a quick glance), I'd rather leave efforts to refactor out of this PR -- if the refactors get merged first, I'll update the approach here.

Reply:
Suhas is right. I think until we move addr data into net_processing there are only bad and less bad approaches here. Leaving this in net to avoid having to make net_processing function calls in ForEachNode() seems like the less bad option to me (see https://github.com/bitcoin/bitcoin/pull/20726#discussion_r575271596).